### PR TITLE
fix(monitors): Fixes multiple check-ins via upsert

### PR DESCRIPTION
**Pull request by rjo100**
Allows customers to upsert for the same monitor with multiple check-ins

Fixes https://github.com/getsentry/team-crons/issues/63

### CODE REVIEW

File: tests/sentry/monitors/endpoints/test_monitor_ingest_checkin_index.py [hunk 1]
```diff
@@ -182,6 +182,44 @@ def test_monitor_creation_via_checkin(self):
             checkins = MonitorCheckIn.objects.filter(monitor=monitor)
             assert len(checkins) == 1
 
+    def test_monitor_environment_creation_via_checkin(self):
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/endpoints/test_monitor_ingest_checkin_index.py:185]: I think this test doesn't actually need to test environments, I think we jsut needed a test that creates two checkins in a row, and ensure the first creates and the second updates.

Comment (rjo100) [reply tests/sentry/monitors/endpoints/test_monitor_ingest_checkin_index.py:185]: Ah you're right it still fails. Will update + rename

**Review by evanpurkhiser — APPROVED** [review]
awesome.

Can we make sure this also works for ingest via the consumer?

**Comment by rjo100** [comment]
as discussed in person - consumer has separate set of validators

**Pull request merged**